arrangehttp/listener.go

```diff
@@ -154,15 +154,25 @@ type DefaultListenerFactory struct {
 // It essentially does the same thing as net/http, but allows the network
 // to be configured externally and ensures that the listen address matches
 // the server address.
-func (f DefaultListenerFactory) Listen(ctx context.Context, server *http.Server) (net.Listener, error) {
+func (f DefaultListenerFactory) Listen(ctx context.Context, server *http.Server) (l net.Listener, err error) {
 	network := f.Network
 	if len(network) == 0 {
 		network = "tcp"
 	}
-
-	l, err := f.ListenConfig.Listen(ctx, network, server.Addr)
-	if err != nil {
-		return nil, err
+	if server.Addr != "" {
+		l, err = f.ListenConfig.Listen(ctx, network, server.Addr)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// if address is not defined use loopback
+		l, err = net.Listen("tcp", "127.0.0.1:0")
```

Comment:
Should this use `":0"` to listen to both/either IPv4 & IPv6?  Or do you only want to listen to explicitly one of the IP spaces?

Reply:
This should explicitly listen to one of the IP spaces otherwise it will bind to all addresses. If the consumer want to overwrite that behaviour then they can provide `:0` i.e same the same effect of an empty string now.

Reply:
Ok, that makes sense.  @johnabass this looks good to me, how about you?

Reply:
I'm not so sure this is a good idea.  The problem is obtaining the actual port the server is listening on when you don't supply a listener.  This is an issue in testing scenarios, which is handled right now by libraries other than `arrange` precisely because `arrangehttp` honors the same semantics as `net/http`: no address means ":80".

Reply:
Sounds good, I like it.